### ISSUE TITLE
denom is not shown for Shielded IBC assets

### ISSUE THREAD

**Issue by 0x4r45h** [anoma/namada#2880]
Namada v0.31.9   

Given I have an IBC token called `transfer/channel-0/nam` :   
Querying transparent account shows denom correctly
```bash
$ namadac balance --owner chain-b-wallet

transfer/channel-0/nam: 66
```
And running `balance` command without the `--owner` flag shows a warning about this token, but still denom is printed correctly   
```
$ namadac balance
Last committed epoch: 1
Token transfer/channel-0/nam
: 66, owned by chain-b-wallet
: 66, owned by tnam1qpy3fwtfyfyhalnzlcglzvyuy20a7uqnds8zd456
Skipping a value for key #tnam1pyqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqej6juv/#tnam1p44wgfjlpru34fxfpn28ltu6enpc5mpk9vnqes8j/minter. Error in decoding: Unexpected length of input
```

But, for the shielded account, the established address of token is shown instead of denom       
```
$ namadac balance --owner shield-1
Last committed epoch: 2
tnam1p44wgfjlpru34fxfpn28ltu6enpc5mpk9vnqes8j : 22
```
it is interesting, after shielding , the output of `namadac balance` changes as below (denom disappeared here too) :    
```
$ namadac balance

Shielded Token tnam1p44wgfjlpru34fxfpn28ltu6enpc5mpk9vnqes8j:
  22, owned by zvknam1qdfc854wqqqqpq9ns8puv8qp3vrt5g9d3uj66s8ll0vv6jwnrsg6k542vkc4qqxhrjmgcf6pxare0lkpjrfuj34sg2dejg6jyxvp28kyjrcjvus2f58p03npdq9c47rlfjjxg9aap59juem40wwgqtttka60r83t6jrm623nvlc4ndzskqyy93t6t0ghxn4wu5jm9n8fscv5gk9y6zx23enar63mphcf3sumtvphw2tyzfgvuyatau8294js7f8a6w8yy7ua737cjvgqe9k9w
Token tnam1p44wgfjlpru34fxfpn28ltu6enpc5mpk9vnqes8j
: 22, owned by masp
: 44, owned by chain-b-wallet
: 66, owned by tnam1qpy3fwtfyfyhalnzlcglzvyuy20a7uqnds8zd456
Skipping a value for key #tnam1pyqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqej6juv/#tnam1p44wgfjlpru34fxfpn28ltu6enpc5mpk9vnqes8j/minter. Error in decoding: Unexpected length of input
```

**Comment by yito88** [comment]
Thank you for reporting it. It should be fixed in #2631.
It was caused by storing the IBC denom with the receiver address even if it's a znam address.
In #2631, when the receiver address is a znam address, the IBC denom is stored with the MASP address.

**Comment by 0x4r45h** [comment]
Nice, I keep this open until #2631 gets merged
